src/app/Classes/LogViewer.php

```diff
@@ -0,0 +1,200 @@
+ *
+ * Credits https://github.com/rap2hpoutre/laravel-log-viewer
+ */
+
+class LogViewer
+{
+    /**
+     * @var string file
+     */
+    private static $file;
+
+    private static $levels_classes = [
+        'debug'     => 'info',
+        'info'      => 'info',
+        'notice'    => 'info',
+        'warning'   => 'warning',
+        'error'     => 'danger',
+        'critical'  => 'danger',
+        'alert'     => 'danger',
+        'emergency' => 'danger',
+        'processed' => 'info',
+    ];
+
+    private static $levels_imgs = [
+        'debug'     => 'info',
+        'info'      => 'info',
+        'notice'    => 'info',
+        'warning'   => 'warning',
+        'error'     => 'warning',
+        'critical'  => 'warning',
+        'alert'     => 'warning',
+        'emergency' => 'warning',
+        'processed' => 'info',
+    ];
+
+    /**
+     * Log levels that are used.
+     *
+     * @var array
+     */
+    private static $log_levels = [
+        'emergency',
+        'alert',
+        'critical',
+        'error',
+        'warning',
+        'notice',
+        'info',
+        'debug',
+        'processed',
+    ];
+
+    const MAX_FILE_SIZE = 52428800;
+
+    /**
+     * @param string $file
+     * @throws \Exception
+     */
+    public static function setFile($file)
+    {
+        $file = self::pathToLogFile($file);
+
+        if (app('files')->exists($file)) {
+            self::$file = $file;
+        }
+    }
+
+    /**
+     * @param string $file
+     *
+     * @throws \Exception
+     *
+     * @return string
+     */
+    public static function pathToLogFile($file)
+    {
+        $logsPath = storage_path('logs');
+
+        if (app('files')->exists($file)) { // try the absolute path
+            return $file;
+        }
+
+        $file = $logsPath.'/'.$file;
+
+        // check if requested file is really in the logs directory
+        if (dirname($file) !== $logsPath) {
+            throw new \Exception('No such log file');
+        }
+
+        return $file;
+    }
+
+    /**
+     * @return string
+     */
+    public static function getFileName()
+    {
+        return basename(self::$file);
+    }
+
+    /**
+     * @return array
+     * @throws \Illuminate\Container\EntryNotFoundException
+     */
+    public static function all()
+    {
+        $log = [];
+
+        $pattern = '/\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\].*/';
+
+        if (!self::$file) {
+            $log_file = self::getFiles();
+            if (!count($log_file)) {
+                return [];
+            }
+            self::$file = $log_file[0];
+        }
+
+        if (app('files')->size(self::$file) > self::MAX_FILE_SIZE) {
+            return;
+        }
+
+        $file = app('files')->get(self::$file);
+
+        preg_match_all($pattern, $file, $headings);
+
+        if (!is_array($headings)) {
+            return $log;
+        }
+
+        $log_data = preg_split($pattern, $file);
```

Comment:
The below probably could be split into a more meaningful and semantic name (because I have no clue what this is meant to do).